lib/resources/FileUploads.js

```diff
@@ -9,14 +10,45 @@ module.exports = StripeResource.extend({
 
   overrideHost: 'uploads.stripe.com',
 
-  requestDataProcessor: function(method, data, headers) {
+  requestDataProcessor: function(method, data) {
     data = data || {};
 
     if (method === 'POST') {
-      return multipartDataGenerator(method, data, headers);
+      return determineSource(data);
     } else {
       return utils.stringifyRequestData(data);
     }
+
+    function determineSource(data) {
```

Comment:
With the approach you've got here, this function does more than just `determineSource` - it `getProcessorForSourceType` - so it probably should be renamed so it's clearer as to what you're getting back.

Reply:
sounds good, i'll rename and update!